pkg/jobs/hive/hive.go

```diff
@@ -329,8 +333,268 @@ func UpgradeCluster(client clientv1.Client, clusterName string, curator *cluster
+		Name:      clusterName + "admack",
+		Namespace: clusterName,
+	}, &ocpConfigView); err != nil {
+		// check if mcv exists before creating
+
+		klog.V(2).Info("Create managedclusterview " + clusterName + "admack")
+		if err := client.Create(context.TODO(), ocpConfigMCV); err != nil {
+			return err
+		}
+	}
+
+	ocpConfigGetErr := errors.New("Failed to get remote admin-ack configmap")
+	resultOCPConfigMCV := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := waitForMCV(client, clusterName+"admack", clusterName, &resultOCPConfigMCV, ocpConfigGetErr); err != nil {
+		return err
+	}
+
+	resultConfigMap := resultOCPConfigMCV.Status.Result
+	configMap := map[string]interface{}{}
+
+	if resultConfigMap.Raw != nil {
+		err := json.Unmarshal(resultConfigMap.Raw, &configMap)
+		utils.CheckError(err)
+	} else {
+		return ocpConfigGetErr
+	}
+
+	// Get clusterversion from managed cluster to initiate upgrade
+	managedclusterview := &managedclusterviewv1beta1.ManagedClusterView{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      clusterName,
+			Namespace: clusterName,
+			Labels: map[string]string{
+				MCVUpgradeLabel: clusterName,
+			},
+		},
+		Spec: managedclusterviewv1beta1.ViewSpec{
+			Scope: managedclusterviewv1beta1.ViewScope{
+				Group:     "config.openshift.io",
+				Kind:      "ClusterVersion",
+				Name:      "version",
+				Namespace: "",
+				Version:   "v1",
+			},
+		},
+	}
+
+	mcview := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &mcview); err != nil {
+
+		klog.V(2).Info("Create managedclusterview " + clusterName)
+		if err := client.Create(context.TODO(), managedclusterview); err != nil {
+			return err
+		}
+	}
+
+	resultmcview := managedclusterviewv1beta1.ManagedClusterView{}
+	if err := waitForMCV(client, clusterName, clusterName, &resultmcview, getErr); err != nil {
+		return err
+	}
+
+	resultClusterVersion := resultmcview.Status.Result
+	clusterVersion := map[string]interface{}{}
+
+	if resultClusterVersion.Raw != nil {
+		err := json.Unmarshal(resultClusterVersion.Raw, &clusterVersion)
+		utils.CheckError(err)
+	} else {
+		return getErr
+	}
+
+	currentOCPVer := clusterVersion["status"].(map[string]interface{})["desired"].(map[string]interface{})["version"].(string)
+	desiredSemVer, err := semver.Make(currentOCPVer)
+	if err != nil {
+		return err
+	}
+
+	ackString := "ack-" + strconv.Itoa(int(desiredSemVer.Major)) + "." + strconv.Itoa(int(desiredSemVer.Minor)) +
+		"-kube-1." + strconv.Itoa(int(desiredSemVer.Minor+14)) + "-api-removals-in-" +
+		strconv.Itoa(int(desiredSemVer.Major)) + "." + strconv.Itoa(int(desiredSemVer.Minor+1))
+
+	cmData := configMap["data"]
+	if cmData != nil {
+		// "kube-1.{{ '%02d' | format((ocp_minor_version.split('.')[1] | int) + 14"
+		cmData.(map[string]interface{})[ackString] = "true"
+	} else {
+		configMap["data"] = map[string]interface{}{
```

Comment:
```suggestion
		configMap["data"] = map[string]string{
```
can we do this? maybe a tiny bit better?